### PR TITLE
Add h2spec client subcommand

### PR DESCRIPTION
**Pull request by frobware**
Add h2spec subcommand.

h2spec is a "Conformance testing tool for HTTP/2 implementations" and is used by the origin tests.

### CODE REVIEW

**Comment by frobware** [comment]
Run multi-stage test e2e-aws-operator - e2e-aws-operator-gather-must-gather container test expand_less | 3m34s

d49/controller-manager?timestamps=true": dial tcp 10.0.179.220:10250: connect: connection refused], skipping gathering namespaces/openshift-cluster-csi-drivers due to error: one or more errors ocurred while gathering pod-specific data for namespace: openshift-cluster-csi-drivers      one or more errors ocurred while gathering container data for pod aws-ebs-csi-driver-node-xlq4t:      [Get "https://10.0.179.220:10250/containerLogs/openshift-cluster-csi-drivers/aws-ebs-csi-driver-node-xlq4t/csi-driver?timestamps=true": dial tcp 10.0.179.220:10250: connect: connection refused, Get "https://10.0.179.220:10250/containerLogs/openshift-cluster-csi-drivers/aws-ebs-csi-driver-node-xlq4t/csi-driver?previous=true&timestamps=true": dial tcp 10.0.179.220:10250: connect: connection refused, Get "https://10.0.179.220:10250/containerLogs/openshift-cluster-csi-drivers/aws-ebs-csi-driver-node-xlq4t/csi-node-driver-registrar?timestamps=true": dial tcp 10.0.179.220:10250: connect: connection refused, Get "https://10.0.179.220:10250/containerLogs/openshift-cluster-csi-drivers/aws-ebs-csi-driver-node-xlq4t/csi-node-driver-registrar?previous=true&timestamps=true": dial tcp 10.0.179.220:10250: connect: connection refused, Get "https://10.0.179.220:10250/containerLogs/openshift-cluster-csi-drivers/aws-ebs-csi-driver-node-xlq4t/csi-liveness-probe?timestamps=true": dial tcp 10.0.179.220:10250: connect: connection refused, Get "https://10.0.179.220:10250/containerLogs/openshift-cluster-csi-drivers/aws-ebs-csi-driver-node-xlq4t/csi-liveness-probe?previous=true&timestamps=true": dial tcp 10.0.179.220:10250: connect: connection refused]]error: gather never finished for pod must-gather-26chw: pods "must-gather-26chw" not found {"component":"entrypoint","error":"wrapped process failed: exit status 1","file":"prow/entrypoint/run.go:80","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2021-04-06T11:55:40Z"} error: failed to execute wrapped command: exit status 1```

/retest

File: test/h2spec/client.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+	flags.Int("max-header-length", 4000, "Maximum length of HTTP header")
+	flags.StringP("junit-report", "j", "", "Path for JUnit test report")
+	flags.BoolP("strict", "S", false, "Run all test cases including strict test cases")
+	flags.Bool("dryrun", false, "Display only the title of test cases")
+	flags.BoolP("tls", "t", false, "Connect over TLS")
+	flags.StringP("ciphers", "c", "", "List of colon-separated TLS cipher names")
+	flags.BoolP("insecure", "k", false, "Don't verify server's certificate")
+	flags.BoolP("verbose", "v", false, "Output verbose log")
+	flags.Bool("version", false, "Display version information and exit")
+	flags.Bool("help", false, "Display this help and exit")
```
Comment (candita) [inline test/h2spec/client.go:47]: `help` and `version` won't ever be used, will they?

Comment (candita) [reply test/h2spec/client.go:47]: Also - where is the `yaml` change for this as an ingress-operator  subcommand?

Comment (frobware) [reply test/h2spec/client.go:47]: > `help` and `version` won't ever be used, will they?

Depends if we ever ask a customer to run this. If so, they will be useful. We can verify which version, and the help explains what the options are.

Comment (frobware) [reply test/h2spec/client.go:47]: > Also - where is the `yaml` change for this as an ingress-operator subcommand?

There's no associated YAML for this. Perhaps I'm missing the question here?

Comment (candita) [reply test/h2spec/client.go:47]: It's been merged now, but I think I was looking for deployment yaml to be updated.  I guess it is not needed.

Comment (sgreene570) [reply test/h2spec/client.go:47]: I believe the h2spec client code is only going to be consumed by tests in origin. So no changes are needed to the ingress-operator or router deployment yaml.

**Pull request merged**